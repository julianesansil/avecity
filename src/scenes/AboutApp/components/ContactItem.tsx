import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Icon } from 'native-base';
import styled from 'styled-components/native';

import SCText from '~/src/components/SCText';

import { colors } from '~/src/styles/theme';

interface Props {
  nameIcon: string;
  contactText: string;

  onPress: () => void;
}

function ContactItem({ nameIcon, contactText, onPress }: Props) {
  return (
    <SCContainer>
      <SCIcon name={nameIcon} />

      <TouchableOpacity onPress={onPress}>
        <SCContactText>{contactText}</SCContactText>
      </TouchableOpacity>
    </SCContainer>
  );
}

const SCContainer = styled.View`
  flex-direction: row;
  align-items: center;
`;

export const SCIcon = styled(Icon)`
  font-size: 20;
  color: ${colors.PURPLE};

  padding-top: 6;
`;

export const SCContactText = styled(SCText)`
  margin-left: 10;
  text-decoration-line: underline;

  margin-top: 6;
`;

export default ContactItem;
