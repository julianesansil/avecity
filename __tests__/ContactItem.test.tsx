import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import ContactItem, {
  SCIcon,
  SCContactText,
} from '~/src/scenes/AboutApp/components/ContactItem';
import { TouchableOpacity } from 'react-native';

describe('ContactItem suit tests', () => {
  let wrapper: any;
  const onPress = jest.fn();

  beforeAll(() => {
    wrapper = renderer.create(
      <ContactItem
        nameIcon="call"
        contactText="(11) 95154-5438"
        onPress={onPress}
      />,
    );
  });

  it('renders correctly and match snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly and check values', () => {
    const instanceRoot = wrapper.root;

    expect(instanceRoot.findByType(SCIcon).props.name).toEqual('call');
    expect(instanceRoot.findByType(SCContactText).props.children).toEqual(
      '(11) 95154-5438',
    );
  });

  it('renders correctly and check onpress', () => {
    const instanceRoot = wrapper.root;

    instanceRoot.findByType(TouchableOpacity).props.onPress();
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
