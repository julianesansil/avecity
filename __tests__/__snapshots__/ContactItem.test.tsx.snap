// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem suit tests renders correctly and match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#4B134F",
            "fontSize": 20,
            "paddingTop": 6,
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„¾
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#65615C",
            "fontFamily": "Raleway-Medium",
            "fontSize": 15,
            "marginLeft": 10,
            "marginTop": 6,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      (11) 95154-5438
    </Text>
  </View>
</View>
`;
